src/common/KokkosKernels_IOUtils.hpp

```diff
@@ -647,130 +268,6 @@ inline void kk_read_3Dview_from_file(idx_array_type &view,
-      srcs[j] = i;
-    }
-  }
-}
-
-template <typename size_type, typename lno_t, typename wt>
-void convert_edge_list_to_csr(lno_t nv, size_type ne, lno_t *srcs, lno_t *dests,
-                              wt *ew, size_type *xadj, lno_t *adj, wt *crs_ew) {
-  std::vector<struct Edge<lno_t, wt>> edges(ne);
-  for (size_type i = 0; i < ne; ++i) {
-    edges[i].src = srcs[i];
-    edges[i].dst = dests[i];
-    edges[i].ew  = ew[i];
-  }
-  std::sort(edges.begin(), edges.begin() + ne);
-
-  size_type eind = 0;
-  for (lno_t i = 0; i < nv; ++i) {
-    (xadj)[i] = eind;
-    while (edges[eind].src == i) {
-      (adj)[eind]     = edges[eind].dst;
-      (*crs_ew)[eind] = edges[eind].ew;
-      ++eind;
-    }
-  }
-  xadj[nv] = eind;
-}
-
-template <typename in_lno_t, typename size_type, typename lno_t>
-void convert_undirected_edge_list_to_csr(lno_t nv, size_type ne, in_lno_t *srcs,
-                                         in_lno_t *dests, size_type *xadj,
-                                         lno_t *adj) {
-  std::vector<struct Edge<lno_t, double>> edges(ne * 2);
-  for (size_type i = 0; i < ne; ++i) {
-    edges[i * 2].src = srcs[i];
-    edges[i * 2].dst = dests[i];
-
-    edges[i * 2 + 1].src = dests[i];
-    edges[i * 2 + 1].dst = srcs[i];
-  }
-#ifdef KOKKOSKERNELS_HAVE_OUTER
-#include <parallel/multiseq_selection.h>
-#include <parallel/multiway_merge.h>
-#include <parallel/merge.h>
-#include <parallel/multiway_mergesort.h>
-  __gnu_parallel::parallel_sort_mwms<false, true, struct Edge<lno_t, double> *>(
-      &(edges[0]), &(edges[0]) + ne * 2,
-      std::less<struct Edge<lno_t, double>>(), 64);
-#else
-  std::sort(edges.begin(), edges.begin() + ne * 2);
-#endif
-
-  size_type eind = 0;
-  for (lno_t i = 0; i < nv; ++i) {
-    (xadj)[i] = eind;
-    while (edges[eind].src == i) {
-      (adj)[eind] = edges[eind].dst;
-      //(*crs_ew)[eind] = edges[eind].ew;
-      ++eind;
-    }
-  }
-  xadj[nv] = eind;
-}
-/*
-
-template <typename lno_t, typename size_type, typename scalar_t>
-void read_graph_src_dst_bin(
-    lno_t *nv, size_type *ne
-    ,size_type **xadj, lno_t **adj, scalar_t **ew,
-    const char *fnameSrc, const char *fnameTarg){
-
-  size_t numEdges = 0;
-  size_t *srcs, *dst; //this type is hard coded
-  buildEdgeListFromBinSrcTarg_undirected(
-      fnameSrc, fnameTarg,
-      &numEdges,
-      &srcs, &dst);
-
-  lno_t num_vertex = 0;
-  for (size_t i = 0; i < numEdges; ++i){
-    if (num_vertex < srcs[i]) num_vertex = srcs[i];
-    if (num_vertex < dst[i]) num_vertex = dst[i];
-  }
-  num_vertex += 1;
-
-  *nv = num_vertex;
-  *ne = numEdges * 2;
-
-  md_malloc<size_type>(xadj, num_vertex + 1);
-  md_malloc<lno_t>(adj, numEdges * 2);
-  convert_undirected_edge_list_to_csr (
-      num_vertex, numEdges,
-      srcs, dst,
-      *xadj, *adj);
-
-  delete [] srcs;
-  delete [] dst;
-}
-*/
-
 template <typename idx, typename wt>
 void write_edgelist_bin(size_t ne, const idx *edge_begins, const idx *edge_ends,
```

Comment:
This function is for a sparse structure too and could be moved.

Reply:
On second thought, this is actually not called anywhere and could be removed

Reply:
Okay, I'll just remove then, or just mark as deprecated.